upload.py

```diff
@@ -35,23 +41,127 @@ def main():
+        else:
+            logging.info(f"Uploading {file_path} to dataset id {args.owid_total_tests}")
+            upload(
+                args.wazimap_endpoint,
+                args.wazimap_token,
+                args.owid_total_tests,
+                file_path
+            )
+
+    # TOTAL CASES
+    if args.owid_total_cases:
+        logging.info("Downloading OWID file")
+        owid.download_file()
+
+        logging.info("Processing total cases")
+        file_path = owid.write_total_cases()
+        logging.info(f"File written to {file_path}")
+
+        if args.dont_upload:
+            logging.info(f"Not uploading {file_path} due to --dont-upload")
+        else:
+            logging.info(f"Uploading {file_path} to dataset id {args.owid_total_cases}")
+            upload(
+                args.wazimap_endpoint,
+                args.wazimap_token,
+                args.owid_total_cases,
+                file_path
+            )
+    
+    # TOTAL DEATHS
+    if args.owid_total_deaths:
+        logging.info("Downloading OWID file")
+        owid.download_file()
+
+        logging.info("Processing total deaths")
+        file_path = owid.write_total_deaths()
+        logging.info(f"File written to {file_path}")
+
+        if args.dont_upload:
+            logging.info(f"Not uploading {file_path} due to --dont-upload")
+        else:
+            logging.info(f"Uploading {file_path} to dataset id {args.owid_total_deaths}")
+            upload(
+                args.wazimap_endpoint,
+                args.wazimap_token,
+                args.owid_total_deaths,
+                file_path
+            )
+
+    # GENDER DATA
+    if args.gh5050_gender_data:
+        logging.info("Downloading GH5050 file")
+        gh5050.download_file()
+
+        logging.info("Processing gender data")
+        file_path = gh5050.write_gender_data()
+        logging.info(f"File written to {file_path}")
+
+        if args.dont_upload:
+            logging.info(f"Not uploading {file_path} due to --dont-upload")
+        else:
+            logging.info(f"Uploading {file_path} to dataset id {args.gh5050_gender_data}")
+            upload(
+                args.wazimap_endpoint,
+                args.wazimap_token,
+                args.gh5050_gender_data,
+                file_path
+            )
+
 
 def upload(wazimap_endpoint, wazimap_token, dataset_id, file_path):
     url = f"{wazimap_endpoint}/api/v1/datasets/{dataset_id}/upload/"
 
-    headers = {'authorization': f"Token {wazimap_token}"}
-    files = {'file': open(file_path, 'rb')}
-    payload = {'update': True, 'overwrite': True}
+    # headers = {'authorization': f"Token {wazimap_token}"}
+    # files = {'file': open(file_path, 'rb')}
+    # payload = {'update': True, 'overwrite': True}
 
-    r = requests.post(url, headers=headers, data=payload, files=files)
-    r.raise_for_status()
+    # r = requests.post(url, headers=headers, data=payload, files=files)
+    # r.raise_for_status()
```

Comment:
Why did you comment this out? The --dont-upload flag is supposed to let you run it without uploading :)